pkg/operator/ceph/cluster/cluster_test.go

```diff
@@ -152,73 +156,101 @@ func TestPreMonChecks(t *testing.T) {
 }
 
 func TestConfigureMsgr2(t *testing.T) {
+	configDir := "rook"
+
 	type fields struct {
-		encryptionExpected   bool
-		compressionExpected  bool
-		disabledBothExpected bool
-		cephVersion          cephver.CephVersion
-		Spec                 *cephv1.ClusterSpec
+		encryptionExpected  bool
+		compressionExpected bool
+		cephVersion         cephver.CephVersion
+		Spec                *cephv1.ClusterSpec
 	}
+
 	tests := []struct {
 		name   string
 		fields fields
 	}{
-		{"default settings", fields{false, false, false, cephver.CephVersion{Major: 15}, &cephv1.ClusterSpec{}}},
-		{"encryption enabled", fields{true, false, false, cephver.CephVersion{Major: 16}, &cephv1.ClusterSpec{Network: cephv1.NetworkSpec{Connections: &cephv1.ConnectionsSpec{Encryption: &cephv1.EncryptionSpec{Enabled: true}}}}}},
-		{"compression enabled old version", fields{false, false, false, cephver.CephVersion{Major: 16}, &cephv1.ClusterSpec{Network: cephv1.NetworkSpec{Connections: &cephv1.ConnectionsSpec{Compression: &cephv1.CompressionSpec{Enabled: true}}}}}},
-		{"compression enabled good version", fields{false, true, false, cephver.CephVersion{Major: 17}, &cephv1.ClusterSpec{Network: cephv1.NetworkSpec{Connections: &cephv1.ConnectionsSpec{Compression: &cephv1.CompressionSpec{Enabled: true}}}}}},
-		{"both enabled", fields{true, true, false, cephver.CephVersion{Major: 17}, &cephv1.ClusterSpec{Network: cephv1.NetworkSpec{Connections: &cephv1.ConnectionsSpec{Encryption: &cephv1.EncryptionSpec{Enabled: true}, Compression: &cephv1.CompressionSpec{Enabled: true}}}}}},
-		{"both disabled", fields{false, false, true, cephver.CephVersion{Major: 17}, &cephv1.ClusterSpec{Network: cephv1.NetworkSpec{Connections: &cephv1.ConnectionsSpec{Encryption: &cephv1.EncryptionSpec{Enabled: false}, Compression: &cephv1.CompressionSpec{Enabled: false}}}}}},
+		{"default settings", fields{false, false, cephver.CephVersion{Major: 15}, &cephv1.ClusterSpec{}}},
```

Comment:
I have intentionally removed a few conditions, since if we go on testing all the conditions it will be a huge test and I guess, we are testing good enough tests here but I'm ready to hear your thoughts.

Reply:
Makes sense to simplify the test with the assimilate-conf approach